subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -431,6 +450,14 @@ public void executeTests() {
 
         TestExecuter testExecuter = createTestExecuter();
 
+        if (testListenerBuildOperationAdapter == null) {
```

Comment:
Currently core doesn't know anything about gradle test infrastructure. moving this into `BuildSessionScopeServices` would need to change that.

Reply:
`testing-base` could have its own `PluginServiceRegistry`

Reply:
I've implemented the TestListenerBuildOperationAdapter as BuildSessionScoped service but kept it as explicit `TestListenerBuildOperationAdapter`. I think since it's misleading as some some TaskListenerInternal implementation currently require manual instantiation (e.g. `TestWorkerProgressListener`. Therefore it's cleaner to me to use explicit `TestListenerBuildOperationAdapter` here